core/trino-spi/src/main/java/io/trino/spi/predicate/Domain.java

```diff
@@ -154,6 +154,11 @@ public boolean isOnlyNull()
         return values.isNone() && nullAllowed;
     }
 
+    public boolean isOnlyNotNull()
```

Comment:
Do we need this `Only` prefix here ? (No changes requested wanted to have a discussion here) 

Cc: @findepi

Reply:
I saw the `isOnlyNull` method which (imo) makes it clear what the method means so I thought to keep the name consistent, i.e. if changed to `isNotNull` it could also imply always true, wdyt?

Reply:
`isOnlyNull` means "only null allowed" (null  is just one possible value)

`isOnlyNotNull` means "only not null allowed", but there is no such single thing as "not null".

to avoid confusion, let's call the new method `isAllExceptNull`

Reply:
> isOnlyNotNull means "only not null allowed", but there is no such single thing as "not null".

Not sure what you mean by this. "not null" means any value but null. 

This method should be named `isNotNull()` (i.e., does this domain represent the set of values that are not null?), and it would be the counterpart of `Domain.notNull(...)`. I don't see what's confusing about that.

Reply:
Thanks! Updated. Should the pr name be changed?

Reply:
To me, Domain is "a set", not "a filter".
`isNotNull` is suitable terminology for a filter, but not for the set
maybe `isAllNonNull` would be suitable for a set